R/seroprevalence_data.R

```diff
@@ -120,4 +120,203 @@ prepare_bin_data <- function(serodata) {
     p_obs_bin_u = .data$Upper
   )
   return(xx)
+}
+
+#' Function that generates the probabilities of being previously exposed to a pathogen.
+#'
+#' @param sim_data A dataframe object containing the following columns:
+#' \tabular{ll}{
+#' \code{birth_year} \tab List of years in which the subjects were borned \cr \tab \cr
+#' \code{tsur} \tab Year of the survey\cr \tab \cr
+#' \code{country} \tab Default to 'none'.\cr \tab \cr
+#' \code{survey} \tab Survey label \cr \tab \cr
+#' \code{age_mean_f} \tab Age \cr \tab \cr
+#' }
+#' @param foi Numeric atomic vector corresponding to the desired Force-of-Infection.
+#' @param seed The seed for random number generation.
+#' @return A simulated list containing a seropositivity distribution by age for given simulated
+#' dataset and desired foi trend.
+#' @examples
+#'\dontrun{
+#'
+#' }
+#' @export
+get_sim_prob <- function(sim_data, foi, seed = 1234) {
+  exposure_ages <- get_exposure_ages(sim_data)
+  exposure_matrix <- get_exposure_matrix(sim_data)
+
+  set.seed(seed = seed)
+  sim_probabilities <- purrr::map_dbl(exposure_ages, ~1-exp(-pracma::dot(exposure_matrix[., ], foi)))
```

Comment:
Addressed by commit [1052161](https://github.com/epiverse-trace/serofoi/commit/1052161c2a83694391199b3545256027915f6ab4).

Reply:
Addressed by commit [3065eb5](https://github.com/epiverse-trace/serofoi/commit/3065eb5f9421e946cef82317e5111c30215662c6). I had to revert commit [271b7ac](https://github.com/epiverse-trace/serofoi/commit/271b7ac6f486363f20e194804b82e03bd9d5324a) because I got confused initially by your comment; the output of this function was the list of probabilities to obtain seropositive cases for each age at the time of the survey (if I'm understanding correctly), not the seropositive cases (which are computed by `get_sim_n_seropositive`).